pkg/conversion/capi2mapi/common.go

```diff
@@ -0,0 +1,150 @@
+/*
+Copyright 2024 Red Hat, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package capi2mapi
+
+import (
+	"encoding/json"
+	"fmt"
+	"strings"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	capiv1 "sigs.k8s.io/cluster-api/api/v1beta1"
+)
+
+// RawExtensionFromProviderSpec marshals the machine provider spec.
+func RawExtensionFromProviderSpec(spec interface{}) (*runtime.RawExtension, error) {
+	if spec == nil {
+		return &runtime.RawExtension{}, nil
+	}
+
+	rawBytes, err := json.Marshal(spec)
+	if err != nil {
+		return nil, fmt.Errorf("error marshalling providerSpec: %w", err)
+	}
+
+	return &runtime.RawExtension{
+		Raw: rawBytes,
+	}, nil
+}
+
+func convertCAPIMachineSetSelectorToMAPI(mapiSelector metav1.LabelSelector) metav1.LabelSelector {
+	capiSelector := mapiSelector
+	capiSelector.MatchLabels = convertCAPILabelsToMAPILabels(mapiSelector.MatchLabels)
+
+	return capiSelector
+}
+
+func convertCAPILabelsToMAPILabels(capiLabels map[string]string) map[string]string {
```

Comment:
Do we want unit tests for these? (forgive me if I've missed them!)

Reply:
The fuzzy tests should fully cover the conversion expectations